src/parse/validator.rs

```diff
@@ -124,7 +126,17 @@ impl<'help, 'app, 'parser> Validator<'help, 'app, 'parser> {
                     return Err(Error::invalid_value(
                         self.p.app,
                         val_str.into_owned(),
-                        &arg.possible_vals,
+                        &arg.possible_vals
+                            .iter()
+                            .filter_map(|value| match value {
+                                ArgValue {
+                                    hidden: false,
+                                    value,
+                                    ..
+                                } => Some(value),
+                                _ => None,
+                            })
```

Comment:
You use the following in a lot of places (sometimes with destructuring):

```rust
|value| if value.get_hidden() {
    None
} else {
    Some(value.get_value())
}
```

Let's create a method to do this in `ArgValue` in the reflection API instead.

Reply:
Is there any naming convention for this in clap?

Reply:
Not sure of a convention but it could be something like `fn get_visible_value(&self) -> Option>&str>`